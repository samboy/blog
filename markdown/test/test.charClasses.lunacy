#!/bin/sh
useUTF8=true
_rem=--[=[
# This is a test script to see the characters that cover various classes

LUNACY=""
if command -v lunacy64 >/dev/null 2>&1 ; then
  LUNACY=lunacy64
elif command -v lua5.1 >/dev/null 2>&1 ; then
  LUNACY=lua5.1
elif command -v lua-5.1 >/dev/null 2>&1 ; then
  LUNACY=lua-5.1
elif command -v lunacy >/dev/null 2>&1 ; then
  LUNACY=lunacy
elif command -v luajit >/dev/null 2>&1 ; then
  LUNACY=luajit # I assume luajit will remain frozen at Lua 5.1
fi
if [ -z "$LUNACY" ] ; then
  echo Please install Lunacy or Lua 5.1
  echo Either the version included with this blog -or- the version at
  echo https://github.com/samboy/lunacy
  echo To compile and install the version of Lunacy with the blog:
  echo
  echo     tar xvJf lunacy-2022-12-06.tar.xz
  echo     cd lunacy-2022-12-06/
  echo     make
  echo     sudo cp lunacy /usr/local/bin/
  exit 1
fi

exec $LUNACY $0 "$@"

# ]=]1
-- This script is written in Lua 5.1

-- This script has been donated to the public domain in 2025 by Sam Trenholme
-- If, for some reason, a public domain declation is not acceptable, it
-- may be licensed under the following terms:

-- Copyright 2025 Sam Trenholme
-- Permission to use, copy, modify, and/or distribute this software for
-- any purpose with or without fee is hereby granted.
-- THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
-- WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES
-- OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
-- ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-- WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
-- ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
-- OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.

function Xascii2utf8(glyph)
  g={}
  g["\1"] = "Á" g["\2"] = "É" g["\3"] = "Í" g["\4"] = "Ñ" g["\5"] = "Ó"
  g["\6"] = "Ú" g["\7"] = "Ü" g["\8"] = "¡" g["\11"] = "—"
  g["\14"] = "«" g["\15"] = "»" g["\16"] = "•" g["\17"] = "á"
  g["\18"] = "é" g["\19"] = "í" g["\20"] = "ñ" g["\21"] = "ó"
  g["\22"] = "ú" g["\23"] = "ü" g["\24"] = "¿" g["\25"] = "‘"
  g["\26"] = "’" g["\28"] = "→" g["\29"] = "“" g["\30"] = "”"
  g["\31"] = "©" g["\127"] = "♥"

  if(g[glyph]) then return g[glyph] end
  if(glyph:byte(1,1) >=32 
     and glyph:byte(1,1) < 127) then return glyph:sub(1,1) end
  return nil
end

-- Yes, Cygwin sometimes still assumes Windows applications output CP437
-- Note that ÁÍÓÚ—‘’“”© do not have CP437 points
-- The workaround to avoid this is to pipe the lunacy output through
-- something or redirect it to a file
function Xascii2cp437(glyph)
  local g={}
  g["\1"] = "A" g["\2"] = "\144" g["\3"] = "I" g["\4"] = "\165"
  g["\5"] = "O" g["\6"] = "U" g["\7"] = "\154" g["\8"] = "\173"
  g["\11"] = "---" g["\14"] = "\174" g["\15"] = "\175" g["\16"] = "\7"
  g["\17"] = "\160" g["\18"] = "\130" g["\19"] = "\161" g["\20"] = "\164"
  g["\21"] = "\162" g["\22"] = "\163" g["\23"] = "\129" g["\24"] = "\168"
  g["\25"] = "'" g["\26"] = "'" g["\28"] = "\26" g["\29"] = '"' 
  g["\30"] = '"' g["\31"] = "(C)" g["\127"] = "\3"
  if(g[glyph]) then return g[glyph] end
  if(glyph:byte(1,1) >=32 
     and glyph:byte(1,1) < 127) then return glyph:sub(1,1) end
  return nil
end

print("#",'Rep','%s','%a','%l','%u','%w')
for a=0,255 do
  local z = string.char(a)
  local y = ''
  if useUTF8 then
    y = Xascii2utf8(z)
  else
    y = Xascii2cp437(z)
  end
  if not y then y = 'no' end
  local s = z:find('%s')
  local al = z:find('%a')
  local l = z:find('%l')
  local u = z:find('%u')
  local w = z:find('%w') 
  print(a,y,s,al,l,u,w)
end
  
