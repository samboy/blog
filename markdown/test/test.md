![blogpic](pics/2024-05-01.jpg)
# Title
## DATE

SUMMARY

### Sub-section title
![widepic](pics/PICTURE.png)
_Picture caption_

Some copy here.

### Footnote test

Text@[^1 Footnote *italics* @https://www.example.com link@](fn:1)
Two footnotes together:@[^2 Footnote 2](fn:2)@,[^3 Footnote 3](fn:3)

@[^4 Footnote 4](fn:4)
@[^5 Footnote 5](fn:5)
@[^6 Footnote 6](fn:6)
@[^7 Footnote 7](fn:7)
@[^8 Footnote 8](fn:8)
@[^9 Footnote 9](fn:9)
@[^10 Footnote 10](fn:10)

The limit is two footnotes together. The footnotes must be numbers
from 1 to 9, so 9 footnotes total. We do not support footnotes where 
footnotes themselves have footnotes, nor footnotes without a reference 
in the main text.

### Xascii

```
  0123456789abcdef
0 .ÁÉÍÑÓÚÜ¡..—..«»
1 •áéíñóúü¿‘’.→“”©
2 .!"#$%&'()*+,-./
3 0123456789:;<=>?
4 @ABCDEFGHIJKLMNO
5 PQRSTUVWXYZ[\]^_
6 `abcdefghijklmno
7 pqrstuvwxyz{|}~♥
```

All non-ASCII: ÁÉÍÑÓÚÜ¡—«»•áéíñóúü¿‘’→“”©♥

### Misc

*italics* [link](https://example.com) 

Superscripts: 17^th 18^th 20^th 4^th etc.

* List item
* Another bullet item

