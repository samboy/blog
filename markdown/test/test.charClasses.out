#	Rep	%s	%a	%l	%u	%w
0	no	nil	nil	nil	nil	nil
1	Á	nil	nil	nil	nil	nil
2	É	nil	nil	nil	nil	nil
3	Í	nil	nil	nil	nil	nil
4	Ñ	nil	nil	nil	nil	nil
5	Ó	nil	nil	nil	nil	nil
6	Ú	nil	nil	nil	nil	nil
7	Ü	nil	nil	nil	nil	nil
8	¡	nil	nil	nil	nil	nil
9	no	1	nil	nil	nil	nil
10	no	1	nil	nil	nil	nil
11	—	1	nil	nil	nil	nil
12	no	1	nil	nil	nil	nil
13	no	1	nil	nil	nil	nil
14	«	nil	nil	nil	nil	nil
15	»	nil	nil	nil	nil	nil
16	•	nil	nil	nil	nil	nil
17	á	nil	nil	nil	nil	nil
18	é	nil	nil	nil	nil	nil
19	í	nil	nil	nil	nil	nil
20	ñ	nil	nil	nil	nil	nil
21	ó	nil	nil	nil	nil	nil
22	ú	nil	nil	nil	nil	nil
23	ü	nil	nil	nil	nil	nil
24	¿	nil	nil	nil	nil	nil
25	‘	nil	nil	nil	nil	nil
26	’	nil	nil	nil	nil	nil
27	no	nil	nil	nil	nil	nil
28	→	nil	nil	nil	nil	nil
29	“	nil	nil	nil	nil	nil
30	”	nil	nil	nil	nil	nil
31	©	nil	nil	nil	nil	nil
32	 	1	nil	nil	nil	nil
33	!	nil	nil	nil	nil	nil
34	"	nil	nil	nil	nil	nil
35	#	nil	nil	nil	nil	nil
36	$	nil	nil	nil	nil	nil
37	%	nil	nil	nil	nil	nil
38	&	nil	nil	nil	nil	nil
39	'	nil	nil	nil	nil	nil
40	(	nil	nil	nil	nil	nil
41	)	nil	nil	nil	nil	nil
42	*	nil	nil	nil	nil	nil
43	+	nil	nil	nil	nil	nil
44	,	nil	nil	nil	nil	nil
45	-	nil	nil	nil	nil	nil
46	.	nil	nil	nil	nil	nil
47	/	nil	nil	nil	nil	nil
48	0	nil	nil	nil	nil	1
49	1	nil	nil	nil	nil	1
50	2	nil	nil	nil	nil	1
51	3	nil	nil	nil	nil	1
52	4	nil	nil	nil	nil	1
53	5	nil	nil	nil	nil	1
54	6	nil	nil	nil	nil	1
55	7	nil	nil	nil	nil	1
56	8	nil	nil	nil	nil	1
57	9	nil	nil	nil	nil	1
58	:	nil	nil	nil	nil	nil
59	;	nil	nil	nil	nil	nil
60	<	nil	nil	nil	nil	nil
61	=	nil	nil	nil	nil	nil
62	>	nil	nil	nil	nil	nil
63	?	nil	nil	nil	nil	nil
64	@	nil	nil	nil	nil	nil
65	A	nil	1	nil	1	1
66	B	nil	1	nil	1	1
67	C	nil	1	nil	1	1
68	D	nil	1	nil	1	1
69	E	nil	1	nil	1	1
70	F	nil	1	nil	1	1
71	G	nil	1	nil	1	1
72	H	nil	1	nil	1	1
73	I	nil	1	nil	1	1
74	J	nil	1	nil	1	1
75	K	nil	1	nil	1	1
76	L	nil	1	nil	1	1
77	M	nil	1	nil	1	1
78	N	nil	1	nil	1	1
79	O	nil	1	nil	1	1
80	P	nil	1	nil	1	1
81	Q	nil	1	nil	1	1
82	R	nil	1	nil	1	1
83	S	nil	1	nil	1	1
84	T	nil	1	nil	1	1
85	U	nil	1	nil	1	1
86	V	nil	1	nil	1	1
87	W	nil	1	nil	1	1
88	X	nil	1	nil	1	1
89	Y	nil	1	nil	1	1
90	Z	nil	1	nil	1	1
91	[	nil	nil	nil	nil	nil
92	\	nil	nil	nil	nil	nil
93	]	nil	nil	nil	nil	nil
94	^	nil	nil	nil	nil	nil
95	_	nil	nil	nil	nil	nil
96	`	nil	nil	nil	nil	nil
97	a	nil	1	1	nil	1
98	b	nil	1	1	nil	1
99	c	nil	1	1	nil	1
100	d	nil	1	1	nil	1
101	e	nil	1	1	nil	1
102	f	nil	1	1	nil	1
103	g	nil	1	1	nil	1
104	h	nil	1	1	nil	1
105	i	nil	1	1	nil	1
106	j	nil	1	1	nil	1
107	k	nil	1	1	nil	1
108	l	nil	1	1	nil	1
109	m	nil	1	1	nil	1
110	n	nil	1	1	nil	1
111	o	nil	1	1	nil	1
112	p	nil	1	1	nil	1
113	q	nil	1	1	nil	1
114	r	nil	1	1	nil	1
115	s	nil	1	1	nil	1
116	t	nil	1	1	nil	1
117	u	nil	1	1	nil	1
118	v	nil	1	1	nil	1
119	w	nil	1	1	nil	1
120	x	nil	1	1	nil	1
121	y	nil	1	1	nil	1
122	z	nil	1	1	nil	1
123	{	nil	nil	nil	nil	nil
124	|	nil	nil	nil	nil	nil
125	}	nil	nil	nil	nil	nil
126	~	nil	nil	nil	nil	nil
127	♥	nil	nil	nil	nil	nil
128	no	nil	nil	nil	nil	nil
129	no	nil	nil	nil	nil	nil
130	no	nil	nil	nil	nil	nil
131	no	nil	nil	nil	nil	nil
132	no	nil	nil	nil	nil	nil
133	no	nil	nil	nil	nil	nil
134	no	nil	nil	nil	nil	nil
135	no	nil	nil	nil	nil	nil
136	no	nil	nil	nil	nil	nil
137	no	nil	nil	nil	nil	nil
138	no	nil	nil	nil	nil	nil
139	no	nil	nil	nil	nil	nil
140	no	nil	nil	nil	nil	nil
141	no	nil	nil	nil	nil	nil
142	no	nil	nil	nil	nil	nil
143	no	nil	nil	nil	nil	nil
144	no	nil	nil	nil	nil	nil
145	no	nil	nil	nil	nil	nil
146	no	nil	nil	nil	nil	nil
147	no	nil	nil	nil	nil	nil
148	no	nil	nil	nil	nil	nil
149	no	nil	nil	nil	nil	nil
150	no	nil	nil	nil	nil	nil
151	no	nil	nil	nil	nil	nil
152	no	nil	nil	nil	nil	nil
153	no	nil	nil	nil	nil	nil
154	no	nil	nil	nil	nil	nil
155	no	nil	nil	nil	nil	nil
156	no	nil	nil	nil	nil	nil
157	no	nil	nil	nil	nil	nil
158	no	nil	nil	nil	nil	nil
159	no	nil	nil	nil	nil	nil
160	no	nil	nil	nil	nil	nil
161	no	nil	nil	nil	nil	nil
162	no	nil	nil	nil	nil	nil
163	no	nil	nil	nil	nil	nil
164	no	nil	nil	nil	nil	nil
165	no	nil	nil	nil	nil	nil
166	no	nil	nil	nil	nil	nil
167	no	nil	nil	nil	nil	nil
168	no	nil	nil	nil	nil	nil
169	no	nil	nil	nil	nil	nil
170	no	nil	nil	nil	nil	nil
171	no	nil	nil	nil	nil	nil
172	no	nil	nil	nil	nil	nil
173	no	nil	nil	nil	nil	nil
174	no	nil	nil	nil	nil	nil
175	no	nil	nil	nil	nil	nil
176	no	nil	nil	nil	nil	nil
177	no	nil	nil	nil	nil	nil
178	no	nil	nil	nil	nil	nil
179	no	nil	nil	nil	nil	nil
180	no	nil	nil	nil	nil	nil
181	no	nil	nil	nil	nil	nil
182	no	nil	nil	nil	nil	nil
183	no	nil	nil	nil	nil	nil
184	no	nil	nil	nil	nil	nil
185	no	nil	nil	nil	nil	nil
186	no	nil	nil	nil	nil	nil
187	no	nil	nil	nil	nil	nil
188	no	nil	nil	nil	nil	nil
189	no	nil	nil	nil	nil	nil
190	no	nil	nil	nil	nil	nil
191	no	nil	nil	nil	nil	nil
192	no	nil	nil	nil	nil	nil
193	no	nil	nil	nil	nil	nil
194	no	nil	nil	nil	nil	nil
195	no	nil	nil	nil	nil	nil
196	no	nil	nil	nil	nil	nil
197	no	nil	nil	nil	nil	nil
198	no	nil	nil	nil	nil	nil
199	no	nil	nil	nil	nil	nil
200	no	nil	nil	nil	nil	nil
201	no	nil	nil	nil	nil	nil
202	no	nil	nil	nil	nil	nil
203	no	nil	nil	nil	nil	nil
204	no	nil	nil	nil	nil	nil
205	no	nil	nil	nil	nil	nil
206	no	nil	nil	nil	nil	nil
207	no	nil	nil	nil	nil	nil
208	no	nil	nil	nil	nil	nil
209	no	nil	nil	nil	nil	nil
210	no	nil	nil	nil	nil	nil
211	no	nil	nil	nil	nil	nil
212	no	nil	nil	nil	nil	nil
213	no	nil	nil	nil	nil	nil
214	no	nil	nil	nil	nil	nil
215	no	nil	nil	nil	nil	nil
216	no	nil	nil	nil	nil	nil
217	no	nil	nil	nil	nil	nil
218	no	nil	nil	nil	nil	nil
219	no	nil	nil	nil	nil	nil
220	no	nil	nil	nil	nil	nil
221	no	nil	nil	nil	nil	nil
222	no	nil	nil	nil	nil	nil
223	no	nil	nil	nil	nil	nil
224	no	nil	nil	nil	nil	nil
225	no	nil	nil	nil	nil	nil
226	no	nil	nil	nil	nil	nil
227	no	nil	nil	nil	nil	nil
228	no	nil	nil	nil	nil	nil
229	no	nil	nil	nil	nil	nil
230	no	nil	nil	nil	nil	nil
231	no	nil	nil	nil	nil	nil
232	no	nil	nil	nil	nil	nil
233	no	nil	nil	nil	nil	nil
234	no	nil	nil	nil	nil	nil
235	no	nil	nil	nil	nil	nil
236	no	nil	nil	nil	nil	nil
237	no	nil	nil	nil	nil	nil
238	no	nil	nil	nil	nil	nil
239	no	nil	nil	nil	nil	nil
240	no	nil	nil	nil	nil	nil
241	no	nil	nil	nil	nil	nil
242	no	nil	nil	nil	nil	nil
243	no	nil	nil	nil	nil	nil
244	no	nil	nil	nil	nil	nil
245	no	nil	nil	nil	nil	nil
246	no	nil	nil	nil	nil	nil
247	no	nil	nil	nil	nil	nil
248	no	nil	nil	nil	nil	nil
249	no	nil	nil	nil	nil	nil
250	no	nil	nil	nil	nil	nil
251	no	nil	nil	nil	nil	nil
252	no	nil	nil	nil	nil	nil
253	no	nil	nil	nil	nil	nil
254	no	nil	nil	nil	nil	nil
255	no	nil	nil	nil	nil	nil
